swap/swap.go

```diff
@@ -364,7 +362,7 @@ func (s *Swap) sendCheque(swapPeer *Peer) error {
 	}
 
 	// reset balance;
-	err = s.resetBalance(peer, int64(cheque.Amount))
+	err = s.resetBalance(peer, int64(cheque.CumulativePayout))
```

Comment:
I don't think that was correct before (as `Amount` also was cumulative, also balance is denominated in honey whereas the payout is in eth) and it's still wrong now. Should we not increase the balance by `cheque.Honey` here instead?

Reply:
If it was incorrect, how come tests don't fail? My comment is actually bad though, this is in `sendCheques`, which would be from the debitor/issuer, which would have a negative balance and thus the amount needs to be **added**, not subtracted.

My question here is this one. Let's assume I have -100 of balance, I send a cheque for 100, then the balance gets back to 0. The cumulative payout is 100 for now. Then I get into a new debt, I get to maybe -100 balance again, issue a new cheque for again 100, now the cumulative payout is 200, but my balance should be reset only by 100. If it is reset with `CumulativePayout`, my balance suddenly gets to +100, no?

Reply:
I think the tests don't fail because the only tests that call this are the ones in `protocol_test.go` and all of them only send at most one cheque at which point the cheque`actualAmount` and the `cumulativePayout` will be the same. Regardless it is definitely the wrong thing because accounting needs to be done in honey. As far as I see it the only correct approach is to increase (not set) the balance by `cheque.Honey`.

Reply:
I extended the `TestTriggerPaymentThreshold` test to trigger a second cheque and as expected it did the wrong thing. Increasing by `cheque.Honey` resulted in the expected behaviour.

Reply:
There is definitely another major issue with accounting. If two separate nodes run their balance tracking already diverges significantly after the first cheque.

Reply:
Never mind. It only looked this way due to further swap accounting happening in the meantime. Both nodes remain in sync now.

Reply:
I think this is the only actual in-process simulation we should write: That we spawn up a number of nodes, exchange some data and then check that all balances are symmetrical.